### ISSUE TITLE
bug: `ion-datetime` mode="ios" swipe locks when swiping next/previous on desktop

### ISSUE THREAD

**Issue by StevieApp** [ionic-team/ionic-framework#24482]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

try date time swiping left or right and the months don't switch but it presents a new set of dates with min and max implemented . But the dates are not clickable. I'm not sure if I'm the only one facing this issue

```
<ion-card class="ion-no-padding">
<ion-card-content class="ion-no-padding">
<ion-datetime>
<div slot="title"> {{name | translate}} </div>
</ion-datetime>
</ion-card-content>
</ion-card>
```

### Expected Behavior

The months should switch. or at least the date should be clickable then it can switch.

### Steps to Reproduce

Just display a simple date time then display try swiping left and right

### Code Reproduction URL

- https://github.com/sean-perkins/gh-24482-date-swipe

**Comment by sean-perkins** [comment]
Hello @StevieApp thanks for reporting this issue.

I've created a reproduction here: https://github.com/sean-perkins/gh-24482-date-swipe. I'll update the title to best describe the issue. 

The problem is that the `ios` approach for datetime is built around limitations with Safari's Intersection Observer implementation. We didn't really intend for developers to use the `ios` display mode for desktop touch devices. 

The intersection observer fires at a 70% threshold, which accounts for some oddities in Safari. We can likely explore additional platform checks to only do this behavior on a mobile device.

**Comment by StevieApp** [comment]
Thank you. That would be great especially if the results can be replicated
to mobile devices i.e. android and iOS.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/24421, and a fix was released as part of Ionic 6.0.5.
